.github/workflows/spelling.yaml

```
@@ -0,0 +1,29 @@
+name: Check Spelling
+
+on: pull_request
+
+jobs:
+  check-spelling:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          # Fetch the entire history of the repository. Attempting to check out
+          # the merge commit and the base branch does not save any time.
+          fetch-depth: 0
+
+      - name: Run spelling check
+        run: |
+          $targetCommittish = git rev-parse origin/${{ github.base_ref }}
+          ./eng/common/scripts/check-spelling-in-changed-files.ps1 `
+              -CSpellConfigPath 'cspell.json' `
+              -SourceCommittish '${{ github.sha }}' `
+              -TargetCommittish  $targetCommittish `
```

```suggestion
        uses: actions/checkout@v2
        with:
          fetch-depth: 2

      - name: Run spelling check
        run: |
          ./eng/common/scripts/check-spelling-in-changed-files.ps1 `
              -CSpellConfigPath 'cspell.json' `
              -SourceCommittish HEAD `
              -TargetCommittish HEAD^ `
```

I think this should work.  It's what we use in TSV.

https://github.com/Azure/azure-rest-api-specs/blob/67c84bd51ccfca841c39f274b9147347abed3f66/.github/workflows/typespec-validation.yaml#L11-L13
https://github.com/Azure/azure-rest-api-specs/blob/67c84bd51ccfca841c39f274b9147347abed3f66/eng/scripts/ChangedFiles-Functions.ps1#L1